src/localstream.js

```diff
@@ -209,25 +241,43 @@ const localstream = (function() {
 		console.log('(ReachSDK::localstream::initAudioVideo)');
 		if (!initAudioVideoProgress) {
 			initAudioVideoProgress=true;
+			_getAllVideoSources();
```

Comment:
_getAllVideoSources is asynchronous so videoSources can be empty if enumerateDevices takes time. Use a Promise base approach